modules/nf-core/hifiasm/main.nf

```diff
@@ -2,17 +2,20 @@ process HIFIASM {
     tag "$meta.id"
     label 'process_high'
 
-    conda "bioconda::hifiasm=0.18.5"
+    conda "bioconda::hifiasm=0.19.7"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/hifiasm:0.18.5--h5b5514e_0' :
-        'quay.io/biocontainers/hifiasm:0.18.5--h5b5514e_0' }"
+        'https://depot.galaxyproject.org/singularity/hifiasm:0.19.7--h43eeafb_0' :
+        'biocontainers/hifiasm:0.19.7--h43eeafb_0' }"
 
     input:
     tuple val(meta), path(reads)
     path  paternal_kmer_dump
     path  maternal_kmer_dump
     path  hic_read1
     path  hic_read2
+    path  (ont_ul, stageAs: "ont_ul.fastq.gz")
+    val   ploidy
+    val   gen_size_kb
```

Comment:
Only required parameters should be supplied through `input:`, and optional parameters come in through `ext.args`. Often the idea is that the data like ploidy and genome size will be part of the meta map, and then you use a closure to supply that value to ext.args:
```
withName: HIFIASM {
    ext.args = { [ 
         meta.ploidy? "--n-hap ${meta.ploidy}" : "",
        gen_size_kb ? "--hg-size ${meta.gen_size_kb}k" : ""
    ].join(' ').trim() }
}
```

Reply:
Yeah, I was thinking about putting them here until we have a solution for the meta map. I'll remove them, but it will be complicated to extract the policy from GOAT and then pass it through an ext.args...

Reply:
```suggestion
```